src/client/datascience/jupyter/kernels/kernel.ts

```diff
@@ -209,6 +179,48 @@ export class Kernel implements IKernel {
             }
         }
     }
+    private async startNotebook(options?: { disableUI?: boolean }): Promise<INotebook> {
+        if (this.restarting) {
+            await this.restarting.promise;
+        }
+        if (!this._notebookPromise) {
+            this.startCancellation = new CancellationTokenSource();
+            this._notebookPromise = new Promise<INotebook>(async (resolve, reject) => {
+                try {
+                    await this.validate(this.uri);
+                    try {
+                        this.notebook = await this.notebookProvider.getOrCreateNotebook({
+                            identity: this.uri,
+                            resource: this.uri,
+                            disableUI: options?.disableUI,
+                            getOnly: false,
+                            metadata: undefined, // No need to pass this, as we have a kernel connection (metadata is required in lower layers to determine the kernel connection).
+                            kernelConnection: this.kernelConnectionMetadata,
+                            token: this.startCancellation.token
+                        });
+                        if (!this.notebook) {
+                            // This is an unlikely case.
```

Comment:
Unlikely, but necessary due to the fact that `getOrCreateNotebook` returns a type of `undefined` (not possible, too messy to fix this typing issue)